server/pulp/server/managers/content/orphan.py

```diff
@@ -231,6 +239,42 @@ def delete_orphans_by_type(content_type_id, content_unit_ids=None):
                 OrphanManager.delete_orphaned_file(storage_path)
 
     @staticmethod
+    def delete_orphan_content_units_by_type(type_id):
+        """
+        Delete the orphaned content units for the given content type.
+        This method only applies to new style content units that are loaded via entry points
+
+        NOTE: this method deletes the content unit's bits from disk, if applicable.
+
+        :param type_id: id of the content type
+        :type type_id: basestring
+        """
+        # get the model matching the type
+        content_model = plugin_api.get_unit_model_by_id(type_id)
+        content_units = content_model.objects().only('id', 'storage_path')
```

Comment:
FYI, you don't need the `()` on `objects()`. This could be written as `content_model.objects.only('id', 'storage_path')`.